### ISSUE TITLE
mswin.vim seems insert vim script randomly with c-s,c-v

### ISSUE THREAD

**Issue by NewUserHa** [vim/vim#4975]
**Describe the bug**
1. c-v the first one https://github.com/dense-analysis/ale/issues/1468
2. c-s I still got randomly 
```
:update
gi
```
if delete it then press c-s again, it is before cusor again.
if keep it then press c-s again, there's no more it and save sccessfully.
3. installed with `left click` -> `visual select` but actually is `select`

1 and 2 are really annoying but not happen every time. 1 feels 10% and 2 feels is 40%. but once encounter, it's >90% keep be there until reopening gvim

the only thing contains the related text is mswin.vim and its dependency. 

**To Reproduce**
1 is detailed in that issue
2 is with autoindent or from line start or select something(w/o mouse) then backspace then c-s
3 always there

**Expected behavior**
c-s save
c-v copy

**Screenshots**
1 has in that issue.
2 
![1](https://user-images.githubusercontent.com/32261870/65569777-56bd2880-df91-11e9-815b-ecb4ac4a6d71.gif)

**Environment (please complete the following information):**
gvim_8.1.1967_x64_signed.exe and gvim81.exe
win10 17134.472 and win 10 17134.950
 - Terminal: use gui

**Additional context**
I could reproduce 2 sometimes with blow min .vimrc:
```
set langmenu=none
call plug#begin('$VIM/vimfiles/plugged')
"Plug 'ervandew/supertab'
Plug 'SirVer/ultisnips'
Plug 'honza/vim-snippets'
call plug#end()
set ignorecase
"" ---------------------------------------------------------------

" Vim with all enhancements
source $VIMRUNTIME/vimrc_example.vim

" Remap a few keys for Windows behavior
source $VIMRUNTIME/mswin.vim

" Use the internal diff if available.
" Otherwise use the special 'diffexpr' for Windows.
if &diffopt !~# 'internal'
  set diffexpr=MyDiff()
endif
function MyDiff()
  let opt = '-a --binary '
  if &diffopt =~ 'icase' | let opt = opt . '-i ' | endif
  if &diffopt =~ 'iwhite' | let opt = opt . '-b ' | endif
  let arg1 = v:fname_in
  if arg1 =~ ' ' | let arg1 = '"' . arg1 . '"' | endif
  let arg1 = substitute(arg1, '!', '\!', 'g')
  let arg2 = v:fname_new
  if arg2 =~ ' ' | let arg2 = '"' . arg2 . '"' | endif
  let arg2 = substitute(arg2, '!', '\!', 'g')
  let arg3 = v:fname_out
  if arg3 =~ ' ' | let arg3 = '"' . arg3 . '"' | endif
  let arg3 = substitute(arg3, '!', '\!', 'g')
  if $VIMRUNTIME =~ ' '
    if &sh =~ '\<cmd'
      if empty(&shellxquote)
        let l:shxq_sav = ''
        set shellxquote&
      endif
      let cmd = '"' . $VIMRUNTIME . '\diff"'
    else
      let cmd = substitute($VIMRUNTIME, ' ', '" ', '') . '\diff"'
    endif
  else
    let cmd = $VIMRUNTIME . '\diff'
  endif
  let cmd = substitute(cmd, '!', '\!', 'g')
  silent execute '!' . cmd . ' ' . opt . arg1 . ' ' . arg2 . ' > ' . arg3
  if exists('l:shxq_sav')
    let &shellxquote=l:shxq_sav
  endif
endfunction
```

please fix this issue which makes ppl out of maid.
thanks

**Comment by NewUserHa** [comment]
add a gif of issue 2. the behavior of issue1 and issue 2 is very similar weirdly.

just found issue can fix in a short time after backing to normal mode in the same vim process/session.

**Comment by chrisbra** [comment]
Is there an easy way to reproduce starting from `vim --clean`?

**Comment by NewUserHa** [comment]
no. `vim --clean` is without `mswin.vim`, right?

**Comment by chrisbra** [comment]
well, then source mswin.vim afterwards `vim --clean -c 'ru! mswin.vim'`

**Comment by chrisbra** [comment]
I have been looking at your gif, but it is not clear to me, what you typed and in what mode vim was. Please also indicate exactly what happens.

**Comment by NewUserHa** [comment]
at start, in insert mode, press c-s.

**Comment by NewUserHa** [comment]
the issue 1 gif is more clear. and the issues are alike

**Comment by chrisbra** [comment]
> at start, in insert mode, press c-s.

I tried: `gvim --clean -c "ru! mswin.vim" test.txt`

Entered insert mode and inserted a space by pressing `i<Space>`. Then pressed `<C-S>`  and it seems to work. Did not insert the mapping content. Do you have somehow paste mode set?

> the issue 1 gif is more clear. and the issues are alike

That one links to a really long issue thread at the ale plugin. And even there it is not clear what exactly is the problem and seems to be unreproducible. So please provide clear exact steps to reproduce.

**Comment by chrisbra** [comment]
It looks like your are using select mode. However even when trying to start select mode, `<c-s>` works here. So I must be missing something.

**Comment by NewUserHa** [comment]
it's not 100% reproducible. there's a chance which is commented in the first comment.

`set paste?` = `nopaste`

I can't guarantee it will happen at some time as well. but it often will happen.

the one in ale issue tracker is a bigger one and easier to reproduce than this one. I'm not 100% sure it will happen without ale, but I just never seen it.
ale's author doesn't know what reason it may be and also guess it may be caused by vim instead of ale.

**Comment by chrisbra** [comment]
That's why I am asking for an easy way to reproduce (preferably without having to install other plugins). It's hard to guess what's happening otherwise. I suppose when ale is running Vim is in a state where the mapping does not do the right thing.

**Comment by NewUserHa** [comment]
Issue 1 has a bigger chance to reproduce than this. but the author of ale said he can't reproduce though.
and I don't have methods to guarantee or improve the chance to reproduce them.

but I can feel/guess the mapping failed in those state(half failed), and lag of vim may cause it(maybe).

if there's anything I can help. I can help.

**Comment by NewUserHa** [comment]
I found the `:update gi` one situation seems has nested insert mode(press once esc still in insert mode, then press again back to normal mode).
`paste#paste` is not

**Comment by chrisbra** [comment]
Do you mean block insert mode?

**Comment by NewUserHa** [comment]
what's it?
I mean "when `:update gi` happen, it's still in insert mode if press `esc` first time. it can go back normal mode if press `esc` 2d time."
but not sure if it's 100% this reason since it's randomly bug.

And I checked insert mode with `:call paste()#paste` can back to normal with only once `esc`

**Comment by chrisbra** [comment]
So how do you do that? I am not sure what you need to do, to have to press ESC twice. Please explain

**Comment by NewUserHa** [comment]
just open vim with `edit with vim` with right-click, press `i` or `ci'` or `a`(maybe), do something, then it just may happen.
when press the first `esc` the text/prompt in command bar will flash as normal so I think the `esc` affect.

all plugins were enableded so don't know if it was the same with the min config commented above.
and I don't remember the phenomenon with issue 1.

of course the `p`, `y` is working without any issue.

**Comment by chrisbra** [comment]
Okay, so can you please give step by step instructions starting from `gvim.exe --clean -c "ru! mswin.vim" filename` that shows the behaviour? It's hard to follow your messages, so please be very specific with what you expect and what you see. Thanks!

**Comment by NewUserHa** [comment]
the issues are not able to reproduce just with `gvim.exe --clean -c "ru! mswin.vim" filename`.

the c-v issue needs `ale`.
the c-s issue needs what I don't know.
